components/match2/commons/match_group_input_util.lua

```diff
@@ -1165,4 +1165,49 @@ function MatchGroupInputUtil.standardProcessMatch(match, Parser, mapProps)
 	return match
 end
 
+---@class MapParserInterface
+---@field calculateMapScore? fun(map: table): fun(opponentIndex: integer): integer?
+---@field getExtraData? fun(match: table, games: table, opponents: table[]): table?
+
+--- It may optionally have the following functions:
+--- - calculateMatchScore(map): fun(opponentIndex): integer?
+--- - getExtraData(match, map, opponents): table?
+---@param match table
+---@param opponents table[]
+---@param Parser MapParserInterface
+---@param mapProps any?
+---@return table
+function MatchGroupInputUtil.standardProcessMaps(match, opponents, Parser, mapProps)
+	local maps = {}
+	for key, map in Table.iter.pairsByPrefix(match, 'map', {requireIndex = true}) do
+		local finishedInput = map.finished --[[@as string?]]
+		local winnerInput = map.winner --[[@as string?]]
+
+		map.extradata = Parser.getExtraData and Parser.getExtraData(match, map, opponents) or nil
+		map.finished = MatchGroupInputUtil.mapIsFinished(map)
+
+		local opponentInfo = Array.map(opponents, function(_, opponentIndex)
+			local score, status = MatchGroupInputUtil.computeOpponentScore({
+				walkover = map.walkover,
+				winner = map.winner,
+				opponentIndex = opponentIndex,
+				score = map['score' .. opponentIndex],
+			}, Parser.calculateMapScore and Parser.calculateMapScore(map) or nil)
+			return {score = score, status = status}
+		end)
+
+		map.scores = Array.map(opponentInfo, Operator.property('score'))
+		if map.finished then
+			map.resulttype = MatchGroupInputUtil.getResultType(winnerInput, finishedInput, opponentInfo)
+			map.walkover = MatchGroupInputUtil.getWalkover(map.resulttype, opponentInfo)
+			map.winner = MatchGroupInputUtil.getWinner(map.resulttype, winnerInput, opponentInfo)
+		end
+
+		table.insert(maps, map)
+		match[key] = nil
+	end
+
+	return maps
+end
```

Comment:
parts of this is oos for sure^^

- maybe split input and return value up? like `mapInput` and `map`
- there seems t be missing quite a few things that would be needed by several wikis
  - `map.map = Parser.getName and Parser.getName(mapInput) or mapInput.name`
  - `map.subgroup = tonumber(mapInput.subgroup) or (subGroup + 1)` plus define `local subgroup = 0` before the loop --> for wikis that do not use it it would just be the mapIndex instead of nil, don't think it would do any harm
  - `map.patch = Parser.getPatch and Parser.getPatch(map, match) or mapInput.patch`
  - `map.vod = Logic.emptyOr(mapInput.vod, match['vodgame' .. mapIndex])`
  - `map.mode = Parser.getMode and Parser.getMode(map) or mapInput.mode`
  - `map.length = mapInput.length`
  - date reading (optional)
  - participants reading
- extradata can only be determined after participants data is known/set on some wikis while others might need it before score processing???

Reply:
1) Out of scope

2) Out of scope for this PR, doing the base case for now

3) Good point, I'll move it last to future proof. No wiki should have a dependency on data in extradata for other stuff, pretty sure the few that were I worked away at the big MGI refactor